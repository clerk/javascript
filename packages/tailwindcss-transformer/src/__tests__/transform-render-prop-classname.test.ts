import { transform } from '../index';

test('render prop with className', () => {
  const result = transform(
    `const Button: React.FC<ButtonProps> = ({ children, className }) => {
    return <div render={() => <button className='bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded'>{children}</button>} />
  }`,
    {
      styleCache: new Map(),
    },
  );
  expect(result).toMatchSnapshot();
});
