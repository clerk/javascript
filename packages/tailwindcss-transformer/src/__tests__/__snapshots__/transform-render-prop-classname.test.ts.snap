// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render prop with className 1`] = `
"const Button: React.FC<ButtonProps> = ({ children, className }) => {
    return <div render={() => <button className="cl-c2153e59">{children}</button>} />;
  }"
`;
